nilearn/glm/_base.py

```diff
@@ -546,6 +536,16 @@ def _generate_statistical_maps(
         # Override automatic detection with explicit type if provided
         stat_type = contrast_types.get(contrast_name, stat_type)
 
+        fields = {
+            "prefix": prefix,
+            "suffix": "statmap",
+            "extension": "nii.gz",
+            "entities": deepcopy(entities),
+        }
+
+        if generate_bids_name:
+            fields["prefix"] = None
```

Comment:
values of fields would be updated later in the loop and lead to non sense on further iterations

Reply:
So IIUC, putting the dictionary in the loop resets all the fields?

Reply:
yup exactly